scrapy/core/engine.py

```diff
@@ -392,8 +395,10 @@ def _download(
         finally:
             self._slot.nextcall.schedule()
 
-    @deferred_f_from_coro_f
-    async def open_spider(
+    def open_spider(self, spider: Spider, close_if_idle: bool = True) -> Deferred[None]:
+        return deferred_from_coro(self.open_spider_async(spider, close_if_idle))
+
+    async def open_spider_async(
         self,
         spider: Spider,
         close_if_idle: bool = True,
```

Comment:
Maybe this would be a good change to apply keyword-only (*spider*) and position-only (*close_if_idle*) logic to the new async method, to make things easier for us in case we ever decide to deprecate some part of the signature.